tsfc/driver.py

```diff
@@ -60,21 +53,17 @@ def compile_form(form, prefix="form", parameters=None):
     return kernels
 
 
-def compile_integral(integral_data, form_data, prefix, parameters):
+def compile_integral(integral_data, form_data, prefix, parameters,
+                     backend="pyop2"):
     """Compiles a UFL integral into an assembly kernel.
 
     :arg integral_data: UFL integral data
     :arg form_data: UFL form data
     :arg prefix: kernel name will start with this string
     :arg parameters: parameters object
+    :arg backend: output format
     :returns: a kernel, or None if the integral simplifies to zero
     """
-    # Remove these here, they're handled below.
```

Comment:
Sorry, haven't noticed of the side effect. Good that your tests spotted it. I had similar problems in FFC. Degrees `[None, "auto", "default", -1, "-1"]` and rules `[None, "auto", "default"]` all mean default. I wanted to be able to pass integral data to `tsfc.driver.compile_integral` without a need to copy it and translate metadata to the form tsfc accepts, i.e., only `"auto"` as degree and scheme.

It would be very helpful if you could support those aliases to `"auto"`. In any case, feel free to fix according to your needs now and I will check later what the consequence to FFC is.